packages/DataTable/stories/examples/Showcase.tsx

```diff
@@ -0,0 +1,127 @@
+import React from "react";
+import StoryHeading from "storybook/components/StoryHeading";
+import { Story, Tagline } from "storybook/assets/styles/common.styles";
+import * as DataTable from "../../src";
+import makeData from "../helpers/makeData";
+
+const ShowcaseStory: () => JSX.Element = () => {
+  const columns = React.useMemo(
+    () => [
```

Comment:
When you depend on the data to make a component, you will get into a problem when you start adding properties like isOpen, isPending, isSelected, isHovered, etc. Because the data has to be mixed with the internal state of the component.

Also, this means that if you want to modify one specific row or cell you will need to mutate the entire data structure to produce a new state in the component, which can be really expensive in big sets of data, this is not noticeable at the beginning but heavily impacts the performance try 3000 rows by 10 columns etc.

This is the reason why people using this approach dont inject the data directly into the component making it control it, instead, they provide a hook where gives you a way to implicitly control the internal state of the table.

Like this on [react-table](https://react-table.tanstack.com/):
  ```jsx
  const {
    getTableProps,
    getTableBodyProps,
    headerGroups,
    rows,
    prepareRow,
  } = useTable({
    columns,
    data,
  })
```
 
 I injected the data in the DataGrid before the creation of hooks and suffered from that heavily, but I didn't attach the state of the component on the data structure like width, or height, those were always separated. 
 
 Maybe you can avoid it in this new component. My two cents from a visitor passing by.